### ISSUE TITLE
building frontend for deployment fails

### ISSUE THREAD

**Issue by lukashartl** [ElektraInitiative/PermaplanT#57]
I created a new branch, which adds build scripts for backend and frontend (for #37), unfortunately building (`vite build` for release) of frontend is failing right now:

```
+ npm run build

> permaplant@0.0.0 build
> tsc && vite build

file:///home/jenkins/workspace/PermaplanT_lh-build-scripts/frontend/node_modules/vite/bin/vite.js:7
    await import('source-map-support').then((r) => r.default.install())
    ^^^^^

SyntaxError: Unexpected reserved word
    at Loader.moduleStrategy (internal/modules/esm/translators.js:133:18)
    at async link (internal/modules/esm/module_job.js:42:21)
npm ERR! code 1
npm ERR! path /home/jenkins/workspace/PermaplanT_lh-build-scripts/frontend
npm ERR! command failed
npm ERR! command sh -c tsc && vite build

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins/.npm/_logs/2023-02-22T22_05_31_393Z-debug.log
```

/cc @markus2330 @aidnurs @buenaflor

**Comment by lukashartl** [comment]
Wait ...

I think this could be related to the nodejs version shipped with Debian... I'll try with the latest node js version and update this issue accordingly.

**Comment by lukashartl** [comment]
It works with NodeJS 18 from NodeSources Repo.
Buildserver has been updated accordingly.
Closing this issue now :)

**Comment by markus2330** [comment]
Just for reference, it also works for me with a older nodejs version:

```
$ npm --version
6.14.18
$ node --version
v14.21.3
$ npm run build
> permaplant@0.0.0 build /home/markus/Projekte/PermaplanT/frontend
> tsc && vite build

vite v3.2.1 building for production...
✓ 508 modules transformed.
dist/index.html                       1.12 KiB
dist/manifest.webmanifest             0.41 KiB
dist/registerSW.js                    0.13 KiB
dist/assets/Root.d960bb53.js          0.24 KiB / gzip: 0.20 KiB
dist/assets/index.8ad4d8ee.js         1.67 KiB / gzip: 0.85 KiB
dist/assets/Root.8eae562e.css         8.83 KiB / gzip: 2.60 KiB
dist/assets/jsx-runtime.5a7de62a.js   139.47 KiB / gzip: 44.89 KiB
dist/assets/App.eaecee78.js           257.18 KiB / gzip: 89.15 KiB

PWA v0.13.1
mode      generateSW
precache  18 entries (493.68 KiB)
files generated
  dist/sw.js
  dist/workbox-7369c0e1.js


   ╭───────────────────────────────────────────────────────────────╮
   │                                                               │
   │      New major version of npm available! 6.14.18 → 9.5.1      │
   │   Changelog: https://github.com/npm/cli/releases/tag/v9.5.1   │
   │               Run npm install -g npm to update!               │
   │                                                               │
   ╰───────────────────────────────────────────────────────────────
```

Which versions did you use?

**Comment by aidnurs** [comment]
Hey guys @markus2330 @lukashartl,

you're both right here:

- as I may recall, the default node version is really old i.e. approx. v8 but needs to be checked
- it should work fine with the version mentioned by Markus

Since we have defined the node version across all the layers of the app, could you please try "our" version i.e. utilized in frontend/backend as well?

https://node.green/

Sincerely,
Nursultan
